clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt

```diff
@@ -160,18 +167,14 @@ interface FossIdRestService {
             logger.info { "The FossID server URL is $url." }
 
             val minReadTimeout = Duration.ofSeconds(60)
-            val retrofit = Retrofit.Builder()
-                .apply {
-                    client(
-                        client?.run {
-                            takeUnless { readTimeoutMillis < minReadTimeout.toMillis() }
-                                ?: newBuilder().readTimeout(minReadTimeout).build()
-                        } ?: OkHttpClient.Builder().readTimeout(minReadTimeout).build()
-                    )
-                }
-                .baseUrl(url)
-                .addConverterFactory(JacksonConverterFactory.create(JSON_MAPPER))
-                .build()
+            val retrofit = Retrofit.Builder().apply {
+                client(
+                    client?.run {
+                        val timeout = maxOf(minReadTimeout, Duration.ofMillis(readTimeoutMillis.toLong()))
+                        newBuilder().readTimeout(timeout).addInterceptor(TimeoutInterceptor).build()
+                    } ?: OkHttpClient.Builder().readTimeout(minReadTimeout).addInterceptor(TimeoutInterceptor).build()
+                )
+            }.baseUrl(url).addConverterFactory(JacksonConverterFactory.create(JSON_MAPPER)).build()
```

Comment:
Can we keep the formatting closer to the original one here to reduce the diff?

Reply:
But actually, I really don't like the complexity / logic duplication in here now. Coming back to [this discussion](https://github.com/oss-review-toolkit/ort/pull/6692/commits/c7fd0332c07181c98b800e038aa77121b7f141ce#r1139953679), shouldn't we really instead unconditionally do `addInterceptor(TimeoutInterceptor)` once for whatever client, and indeed be explicit about all non-default read timeouts by annotating all API endpoints?

I'm also curious about @fviernau's opinion here.

Reply:
> But actually, I really don't like the complexity / logic duplication in here now. Coming back to [this discussion](https://github.com/oss-review-toolkit/ort/pull/6692/commits/c7fd0332c07181c98b800e038aa77121b7f141ce#r1139953679), shouldn't we really instead unconditionally do `addInterceptor(TimeoutInterceptor)` once for whatever client, and indeed be explicit about all non-default read timeouts by annotating all API endpoints?
> 
> I'm also curious about @fviernau's opinion here.

I agree with doing this unconditionally on the client and then annotating the problematic API calls which IIUC are just `readScan` and `deleteScan`. 

I believe moving this to all http client's used in ORT can and should be considered out of scope of this PR, e.g. I would investigate first if there could be nicer annotations 
- like the "@Timeout" one
- how it could work together with setting a value from OrtConfiguration
- maybe a bit more intense testing
- if it applies to ORT at all, because IIUC it was said that these project under `clients` are not really part of ORT but could be considered something separate. This code also dosn't use ORT's `OkHttpClientHelper` but `OkHttp` directly.

Reply:
> I agree with doing this unconditionally on the client and then annotating the problematic API calls which IIUC are just `readScan` and `deleteScan`.

Not that (as [quoted earlier](https://github.com/oss-review-toolkit/ort/pull/6692/commits/c7fd0332c07181c98b800e038aa77121b7f141ce#r1139976953)) at least https://github.com/oss-review-toolkit/ort/commit/034eba71f22d391efde8ac3551ffbc7a0d495801 claims that the high default read timeout value is need for all API calls. Maybe @MarcelBochtler remembers the details.

> like the "@timeout" one

I don't think that's feasible as that's not a built-in annotation that Retrofit provides. Adding a custom annotation instead of this "header trick" would be much more work.

> how it could work together with setting a value from OrtConfiguration

Values for annotations have to be compile-time constants, so that's going to be hard.

Reply:
> default read timeout value is need for all API calls

thanks for pointing this out.

> I don't think that's feasible as that's not a built-in annotation that Retrofit provides. Adding a custom annotation instead of this "header trick" would be much more work.

If the full solution would be like the following one, then it would not be that much work:
https://github.com/square/retrofit/issues/2982#issuecomment-529268601

anyhow, still believe it can be out of scope.

Reply:
> If the full solution would be like the following one

That's indeed nice.

Reply:
> > default read timeout value is need for all API calls
> 
> thanks for pointing this out.
> 
> > I don't think that's feasible as that's not a built-in annotation that Retrofit provides. Adding a custom annotation instead of this "header trick" would be much more work.
> 
> If the full solution would be like the following one, then it would not be that much work: [square/retrofit#2982 (comment)](https://github.com/square/retrofit/issues/2982#issuecomment-529268601)
> 
> anyhow, still believe it can be out of scope.

Yes but this solution relies on reflection. The solution in current PR is based on headers.

Reply:
Nevertheless I will change the logic to do it unconditionally .

Making it global for ORT is, as Frank said, out of scope.